### ISSUE TITLE
Improvements to Site Title Block

### ISSUE THREAD

**Issue by priethor** [WordPress/gutenberg#30918]
**High Priority (target WP 5.8)**
- [x] Whole title styling options:
  - [x] Bold, Italic (#31734)
  - [x] Letter-spacing (#31118)
  - [x] Margin (depends on #28356, fixed in #31728)
  - [x] Padding
  - [x] Font-size and line-height #23228, #29622
- [ ] Make the Site Title icon more understandable #30233
- [ ] Add option not to link to the site home in the frontend #31540
- [x] Normalize toolbar as explained in #25983 (PR: https://github.com/WordPress/gutenberg/pull/30999)

**Low Priority**
- [x] Full feature parity with Heading Block, with fine-grained styling options that require rich text #27830 #30727
- [ ] Add a link to the block description instructions to go to the described location. #30871
- [ ] Ability to prepend/append text to the title, clarify consequences of changing the site title #30730
- [ ] Encourage users to make proper semantic decisions #29644

**Comment by mtias** [comment]
> Full feature parity with Heading Block, with fine-grained styling options that require rich text

I don't think this can be achieved without some significant changes that could be considered breaking ones for the site title.

**Comment by kjellr** [comment]
The Site Title block is also missing Bold and Italic options. These seem expected, considering the block's similarity to the Heading block. Would it be possible to add these to the list?

Site Title Block|Heading Block
---|---
<img width="639" alt="Screen Shot 2021-05-04 at 11 36 14 AM" src="https://user-images.githubusercontent.com/1202812/117029903-1ecda980-accd-11eb-85b2-6251d795dcc1.png">|<img width="573" alt="Screen Shot 2021-05-04 at 11 37 12 AM" src="https://user-images.githubusercontent.com/1202812/117029906-1ffed680-accd-11eb-9bcb-8c809dafd715.png">

**Comment by priethor** [comment]
Thanks, added them to the list.

However, the site title is expected to be plain text, whereas enabling `core/bold` and `core/italic` styles in the Site Title block's RichText generates HTML markup, which would require significant changes as [commented above](https://github.com/WordPress/gutenberg/issues/30918#issuecomment-822279193) or [seen recently with the Post Title block](https://github.com/WordPress/gutenberg/pull/31400#discussion_r625194673). A solution for this would be to enable Bold and Italic formatting as block attributes applying to the whole title.

Would that make sense in the context of the patterns you are working on, @kjellr ?

**Comment by kjellr** [comment]
Yeah, that makes sense technically. Adding them as block attributes to start with might be fine, but I do think it would be worth figuring out a way around that eventually. From a user's perspective, they just want the title to be bold — and they'll be looking for those controls in the same place they look for them for other blocks.

**Comment by carolinan** [comment]
Margin has been added with https://github.com/WordPress/gutenberg/pull/31728

**Comment by youknowriad** [comment]
Let's continue improving the blocks but I'm removing this from the 5.8 board as we're passed the feature freeze date. Bug fixes can continue to land though and important iterations (case by case) can be considered.

**Comment by priethor** [comment]
Let's close this issue as most of the high-priority items are already covered and this is not a focus anymore.
